src/app/shared/components/template/processors/item.ts

```diff
@@ -1,27 +1,28 @@
-// NOTE - importing from 'shared' will fail as contains non-browser packages and
-// name conflicts with local 'shared' folder. Import full path from packages instead
-import { JSEvaluator } from "packages/shared/src/models/jsEvaluator/jsEvaluator";
 import { TemplatedData } from "packages/shared/src/models/templatedData/templatedData";
-
-import { shuffleArray } from "src/app/shared/utils";
 import { FlowTypes } from "../models";
-import { objectToArray } from "../utils";
+import { ItemDataPipe } from "./itemPipe";
+
+type IItemEvalContext = FlowTypes.TemplateRowItemEvalContextMetadata;
+
+interface ITemplateRowWithItemContext extends FlowTypes.TemplateRow {
+  _evalContext: { itemContext: IItemEvalContext }; // force specific context variables when calculating eval statements (such as loop items)
+}
 
 export class ItemProcessor {
   constructor(
-    private dataList: any,
-    private parameterList?: any
+    private dataList: FlowTypes.Data_listRow[] = [],
```

Comment:
Could this property be renamed to indicate more clearly that it represents an array of rows?

Reply:
Good idea, I can't tell you how unnecessarily confusing it's been having ambiguous names without type definitions or tests in the code. If you have at least 1 you can usually infer a certain amount, but definitely good to aim for all 3 where possible. Updated with 569234b